internal/testutils/helpers.go

```diff
@@ -13,3 +14,22 @@ func SkipIfNotRunningIntegrationTests(t *testing.T) {
 		t.Skip("Skipping integration test")
 	}
 }
+
+func GetSourcesCountWithAppType(appTypeId int64) int {
+	var sourcesCount int
+	var sourceIdList []int64
+
+	for _, app := range fixtures.TestApplicationData {
+		if app.ApplicationTypeID == appTypeId {
+			sourceIdList = append(sourceIdList, app.SourceID)
+		}
+	}
+	for _, sourceID := range sourceIdList {
+		for _, src := range fixtures.TestSourceData {
+			if src.ID == sourceID {
+				sourcesCount++
+			}
+		}
+	}
```

Comment:
nice catch, thanks ... it is exactly as you write ... I wanted to count sources that have apps with specific app type id and because before we had in fixtures "invalid" test data (both applications had same app type and same source id) I wanted to find only unique source ids ... but then I changed fixtures because I needed more test data for other tests and I ended with this function above in some "in progress" state and forgot to change it ... so simply ... yes, you are right, it is nonsense and need to be changed as you suggested 😁 😁 😁

Reply:
What I still don't clearly see is if you still want to distinctly count the sources? Because the above function would still return 3 if a source has 3 applications with an specific application type you're looking for.

Reply:
But 1 source can have only 1 app with 1 specific app type. You cannot have 2 apps with same app type for 1 source. So I count with that data in fixtures are valid in this way. Or you mean that we need check that our test data are valid and we dont 2 apps with same app type for 1 source?

Reply:
Well... this might be my misunderstanding. I did remember that I was able to create multiple applications with the same type for the same source, but as you say, this might not make sense and we might need to restrict it somehow. I'm going to ping @lpichler and @lindgrenj6 to help us out here :)

Reply:
Well there is a story around ... we have ticket for it [RHCLOUD-18437](https://issues.redhat.com/browse/RHCLOUD-18437) and I started to solve it but then @lindgrenj6 told me that we need add sql constraint and because it was before your "migrations" PR was merged, it remained unchanged in backlog

Reply:
Aaaha! I see. Thank you for that link :) Everything's clear now.